src/Playwright/API/Generated/IRequest.cs

```diff
@@ -91,9 +99,27 @@ public partial interface IRequest
         /// <summary><para>Returns the <see cref="IFrame"/> that initiated this request.</para></summary>
         IFrame Frame { get; }
 
-        /// <summary><para>An object with HTTP headers associated with the request. All header names are lower-case.</para></summary>
+        /// <summary>
+        /// <para>
+        /// **DEPRECATED** Incomplete list of headers as seen by the rendering engine. Use <see
+        /// cref="IRequest.AllHeadersAsync"/> instead.
+        /// </para>
+        /// </summary>
         Dictionary<string, string> Headers { get; }
 
+        /// <summary>
+        /// <para>
+        /// An array with all the request HTTP headers associated with this request. Unlike
+        /// <see cref="IRequest.AllHeadersAsync"/>, header names are NOT lower-cased. Headers
+        /// with multiple entries, such as <c>Set-Cookie</c>, appear in the array multiple times.
+        /// </para>
+        /// </summary>
+        Task<IReadOnlyList<RequestHeadersArrayResult>> HeadersArrayAsync();
```

Comment:
RequestHeadersArrayResult - this name is confusing - it is not an array. We already have a NameValueEntry, should we get a public type for it?

Reply:
IMHO, the _right_ thing in that case, if we don't want to use the auto-generated type, is to use `KeyValuePair<string, string>`. I had that initially, but wasn't sure if I should keep it.